File: langs/python3/problems/bank_system/stub.py
class Simulation:
    def __init__(self):
        pass

    def create_account(self, timestamp, account_id):
        """Create an account. True if created, false if the id exists."""
        raise NotImplementedError

    def deposit(self, timestamp, account_id, amount):
        """Add funds. New balance, or None if the account is missing."""
        raise NotImplementedError

    def transfer(self, timestamp, source_account_id, target_account_id, amount):
        """Move funds. Source balance, or None if missing, same, or broke."""
        raise NotImplementedError

    def top_spenders(self, timestamp, n):
        """Top accounts as id(outgoing). Incoming does not count.

        Example: ["acc1(500)", "acc2(0)"].
        """
        raise NotImplementedError

    def pay(self, timestamp, account_id, amount):
        """Withdraw now. Returns paymentN, or None.

        Example: "payment1".
        """
        raise NotImplementedError

    def get_payment_status(self, timestamp, account_id, payment):
        """IN_PROGRESS or CASHBACK_RECEIVED, or None if wrong."""
        raise NotImplementedError

    def merge_accounts(self, timestamp, account_id_1, account_id_2):
        """Merge account_id_2 into account_id_1, then delete account_id_2.

        Moves balance, outgoing, payments, history, and pending cashbacks.
        False if either id is missing or they are the same.
        """
        raise NotImplementedError

    def get_balance(self, timestamp, account_id, time_at):
        """Balance at time_at after cashbacks due at timestamp."""
        raise NotImplementedError

File: langs/java/problems/bank_system/stub.java
import java.util.List;

public class Simulation {
    public Simulation() {}

    /**
     * Create an account. Returns true if created, false if the id exists.
     */
    public boolean createAccount(int timestamp, String accountId) {
        return false;
    }

    /**
     * Add funds. Returns the new balance, or null if the account is missing.
     */
    public Integer deposit(int timestamp, String accountId, int amount) {
        return null;
    }

    /**
     * Move funds. Returns the source balance, or null if missing, same
     * account, or insufficient funds.
     */
    public Integer transfer(
            int timestamp, String sourceAccountId, String targetAccountId, int amount) {
        return null;
    }

    /**
     * Top accounts as {@code id(outgoing)}. Incoming does not count.
     * Zero-spend rows appear. Example: {@code ["acc1(500)", "acc2(0)"]}.
     */
    public List<String> topSpenders(int timestamp, int n) {
        return null;
    }

    /**
     * Withdraw now. Returns {@code paymentN}, or null if missing or
     * insufficient. Cashback is 2 percent after 86400000. Example:
     * {@code "payment1"}.
     */
    public String pay(int timestamp, String accountId, int amount) {
        return null;
    }

    /**
     * Returns {@code IN_PROGRESS} or {@code CASHBACK_RECEIVED}, or null
     * if the account or payment is wrong.
     */
    public String getPaymentStatus(int timestamp, String accountId, String payment) {
        return null;
    }

    /**
     * Merge {@code accountId2} into {@code accountId1}, then delete
     * {@code accountId2}. Moves balance, outgoing, payments, history,
     * and pending cashbacks. Returns false if either id is missing or
     * they are the same.
     */
    public boolean mergeAccounts(int timestamp, String accountId1, String accountId2) {
        return false;
    }

    /**
     * Balance at {@code timeAt} after cashbacks due at {@code timestamp}.
     */
    public Integer getBalance(int timestamp, String accountId, int timeAt) {
        return null;
    }
}
